src/main/scala/com/databricks/spark/xml/util/XmlFile.scala

```diff
@@ -82,6 +83,9 @@ private[xml] object XmlFile {
       val writer = new CharArrayWriter()
       val xmlWriter = factory.createXMLStreamWriter(writer)
       val indentingXmlWriter = new IndentingXMLStreamWriter(xmlWriter)
+      if (options.version != null){
+        indentingXmlWriter.writeStartDocument(options.charset, options.version)
```

Comment:
Don't we need to call `writeEndDocument`?

Reply:
yes , we don't need

Reply:
Why?

Reply:
You can have a try with my test class: EmptyElementTest

Reply:
No, I mean why we don't need to call end document while we call start document. We should know why. Is it proper way to write the version?

Reply:
Calling writeEndDocument is a good practice,  but It works without it.   I am not sure if putting the writeEndDocument here is ok?

XmlFile.scala
```
            if (!firstRow) {
              lastRow = false
              indentingXmlWriter.writeEndElement()
              indentingXmlWriter.close()
              writer.toString
            } else {
              indentingXmlWriter.writeEndDocument()
              // This means the iterator was initially empty.
              firstRow = false
              lastRow = false
              ""
            }
```

Reply:
Can we just directly write out at `writer: CharArrayWriter ` at the first place?

Reply:
I meant ... 

```scala
...
val writer = new CharArrayWriter()
val xmlWriter = factory.createXMLStreamWriter(writer)
val indentingXmlWriter = new IndentingXMLStreamWriter(xmlWriter)
writer.write(s"${options.version}\n")
indentingXmlWriter.setIndentStep(indent)

new Iterator[String] {
...
```

Reply:
oh, I see.   I think that's also fine